lib/NormalModuleFactory.js

```diff
@@ -621,19 +628,20 @@ class NormalModuleFactory extends ModuleFactory {
 		);
 	}
 
-	getParser(type, parserOptions) {
-		let ident = type;
-		if (parserOptions) {
-			if (parserOptions.ident) {
-				ident = `${type}|${parserOptions.ident}`;
-			} else {
-				ident = JSON.stringify([type, parserOptions]);
-			}
+	getParser(type, parserOptions = EMPTY_RESOLVE_OPTIONS) {
+		if (!this.parserCache.has(type)) {
```

Comment:
I dislike the double `has` `get` cycle. That kind of duplicate work.

Could you change it to:

``` js
let cache = this.parserCache.get(type);
if (cache === undefined) {
  cache = new WeakMap();
  this.parserCache.set(type, cache);
}
```

Similar for all other cache accesses
```